docs/devguide/content_subscription_profile.rst

```diff
@@ -0,0 +1,547 @@
+
+Filtering and Dispose/Unregister Samples
+========================================
+
+..
+    Sect<5.2.3>
+
+DataReaders without filtering can see samples with the ``valid_data`` field of SampleInfo set to false.
+This happens when the matching DataWriter disposes or unregisters the instance.
+Content filtering (whether achieved through Content-Filtered Topics, Query Conditions, or Multi Topics) will filter such samples when the filter expression explicitly uses key fields.
+Filter expressions that don’t meet that criteria will result in no such samples passing the filter.
+
+.. _content_subscription_profile--content-filtered-topic-example:
+
+Content-Filtered Topic Example
+==============================
+
+..
+    Sect<5.2.4>
+
+The code snippet below creates a content-filtered topic for the Message type.
+First, here is the IDL for Message:
+
+.. code-block:: omg-idl
+
+      module Messenger {
+        @topic
+        struct Message {
+          long id;
+        };
+      };
+
+Next we have the code that creates the data reader:
+
+.. code-block:: cpp
+
+      CORBA::String_var type_name = message_type_support->get_type_name();
+      DDS::Topic_var topic = dp->create_topic("MyTopic",
+                                              type_name,
+                                              TOPIC_QOS_DEFAULT, 0, 0);
+      DDS::ContentFilteredTopic_var cft =
+        participant->create_contentfilteredtopic("MyTopic-Filtered",
+                                                 topic,
+                                                 "id > 1",
+                                                 StringSeq());
+      DDS::DataReader_var dr =
+        subscriber->create_datareader(cft,
+                                      DATAREADER_QOS_DEFAULT, 0, 0);
+
+The data reader ‘``dr``’ will only receive samples that have values of ‘``id``’ greater than 1.
+
+.. _content_subscription_profile--query-condition:
+
+***************
+Query Condition
+***************
+
+..
+    Sect<5.3>
+
+The query condition interface inherits from the read condition interface, therefore query conditions have all of the capabilities of read conditions along with the additional capabilities described in this section.
+One of those inherited capabilities is that the query condition can be used like any other condition with a wait set (see Section :ref:`conditions_and_listeners--conditions`).
+
+The ``DataReader`` interface contains operations for creating (``create_querycondition``) and deleting (``delete_readcondition``) a query condition.
+Creating a query condition requires the following parameters:
+
+* Sample, view, and instance state masksThese are the same state masks that would be passed to ``create_readcondition()``, ``read()``, or ``take()``.
+
+* Query expressionAn SQL-like expression (see :ref:`content_subscription_profile--query-expressions`) describing a subset of samples which cause the condition to be triggered.
```

Comment:
There are a few instances of this where the source document has a bulleted list of terms and their definitions.  In the rst file there's no separator between the term and its definition.

Reply:
Yes, like the footnotes, which I thought I fixed, but guess I didn't, the converter script will just insert the text as is if it doesn't know what to do with something. I will look into fixing both.